e2e/rook-test-framework/rook-infra-manager/rook_test_infra_manager.go

```diff
@@ -0,0 +1,323 @@
+package rook_infra_manager
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/rook/rook/e2e/rook-test-framework/contracts"
+	"github.com/rook/rook/e2e/rook-test-framework/enums"
+	"github.com/rook/rook/e2e/rook-test-framework/objects"
+	"github.com/rook/rook/e2e/rook-test-framework/transport"
+	"github.com/rook/rook/e2e/rook-test-framework/utils"
+	"io/ioutil"
+	"strings"
+	"sync"
+	"time"
+	"runtime"
+	"path"
+)
+
+type rookTestInfraManager struct {
+	transportClient contracts.ITransportClient
+	platformType    enums.RookPlatformType
+	dockerized      bool
+	dockerContext   *objects.DockerContext
+	k8sVersion      enums.K8sVersion
+}
+
+var (
+	r    *rookTestInfraManager
+	once sync.Once
+	curdir string
+)
+
+const (
+	rookOperatorFileName = "rook-operator.yaml"
+	rookClusterFileName = "rook-cluster.yaml"
+	rookClientFileName = "rook-client.yml"
+	podSpecPath = "../../../demo/kubernetes"
+	scriptsPath = "scripts"
+	k8sFalsePostiveSuccessErrorMsg = "exit status 1"
+	rookDindK8sClusterScriptv1_5 = "rook-dind-cluster-v1.5.sh"
+	rookDindK8sClusterScriptv1_6 = "rook-dind-cluster-v1.6.sh"
+	rookOperatorImagePodSpecTag = "quay.io/rook/rookd:master-latest"
+	rookOperatorCreatedTpr = "cluster.rook.io"
+)
+
+func GetRookTestInfraManager(platformType enums.RookPlatformType, isDockerized bool, version enums.K8sVersion) (error, *rookTestInfraManager) {
```

Comment:
idiomatic golang always has the ordering of return values such that error is the last value.  First should be your useful return values then last is always error.  So for this function, the return should be `(*rookTestInfraManager, error)`